framework/src/main/java/org/checkerframework/framework/type/ElementAnnotationApplier.java

```diff
@@ -69,6 +69,17 @@ public static void apply(
             final AnnotatedTypeMirror type,
             final Element element,
             final AnnotatedTypeFactory typeFactory) {
+        applyInternal(type, element, typeFactory);
+        // Also copy annotations from type parameters to their uses.
+        typeFactory.typeVarAnnotator.visit(type);
```

Comment:
Couldn't the conditions from `TypeVarAnnotator` be merged into `applyInternal`?
It would be nice to understand and document when `TypeVarAnnotator` should be used and when `TypeVarUseApplier`.

Reply:
Further to our in-person discussion: e.g. `TypeVarUseApplier` internally also creates a new visitor instance. So doing this for the `TypeVarAnnotator` doesn't seem too bad.
Eventually however we should try to merge those two.